workflow-templates/im-deploy-on-prem-website.yml

```diff
@@ -0,0 +1,460 @@
+      - name: Zip the published app for faster deployment and uploads
+        working-directory: ${{ env.PROJECT_ROOT }}
+        run: |
+          (cd published_app && zip -r ../${{ env.DEPLOY_ZIP }} .)
+
+      # Uncomment this if the site hasn't been created before
+      # This action does work better if the website cert is already installed
+      # - name: Create Web Site
+      #   uses: im-open/iis-site-create@v1.0.0
+      #   with:
+      #     server: '${{ needs.set-vars.outputs.IIS_SERVER }}'
+      #     website-name: '${{ needs.set-vars.outputs.WEBSITE_NAME }}'
+      #     app-pool-name: '${{ needs.set-vars.outputs.APP_POOL_NAME }}'
+      #     website-host-header: '${{ needs.set-vars.outputs.WEBSITE_HOST_HEADER }}'
+      #     website-path: '${{ needs.set-vars.outputs.WEBSITE_PATH }}'
+      #     website-cert-path: '${{ needs.set-vars.outputs.WEBSITE_CERT_PATH }}'
+      #     website-cert-friendly-name: '${{ needs.set-vars.outputs.WEBSITE_CERT_FRIENDLY_NAME }}'
+      #     website-cert-password: '${{ needs.set-vars.outputs.WEBSITE_CERT_PASSWORD }}'
+      #     service-account-id: '${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}'
+      #     service-account-password: '${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}'
+      #     server-public-key: ${{ needs.set-vars.outputs.IIS_SERVER_CERT_PATH }}
+
+      # TODO:  If config transformation is needed for different environments include environment name
+      # For example to perform transformations on a web.config using the web.production.config file:
+      # dotnet publish --configuration Release /p:EnvironmentName=Production
+      - name: Build and Publish
+        working-directory: ${{env.PROJECT_ROOT}}
+        run: |
+          dotnet build --configuration Release
+          dotnet publish -c Release -o ./published_app --no-restore
+
+      - name: Zip the published site for faster deployment and uploads
+        working-directory: ${{env.PROJECT_ROOT}}
+        # TODO:  If you plan on uploading artifacts, add this line to the run:
+        # (cd published_app && zip -r ../${{env.ARTIFACT_ZIP}} . -x */appsettings*.json appsettings*.json)
+        run: |
+          (cd published_app && zip -r ../${{env.DEPLOY_ZIP}} .)
+
+      - name: Stop App Pool
+        id: app-pool-stop
+        if: always()
+        runs-on: [self-hosted, windows-2019]
+        uses: im-open/app-pool-action@v1.0.0
+        with:
+          action: 'stop'
+          server: ${{ needs.set-vars.outputs.IIS_SERVER }}
+          app-pool-name: ${{ needs.set-vars.outputs.APP_POOL_NAME }}
+          service-account-id: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_USER }}
+          service-account-password: ${{ needs.set-vars.outputs.IIS_SERVER_SERVICE_PASSWORD }}
+          server-public-key: ${{ needs.set-vars.outputs.IIS_SERVER_CERT_PATH }}
+
+      - name: Deploy deployment package
+        if: always() && steps.iis-stop.outcome == 'success'
```

Comment:
Why are we using always() here?  The only thing we might want to continue even if it failed is stopping the app pool (but we don't because this step is checking if that one succeeded).  You don't need the always() or the outcome check.  Just let it fail and let the workflow naturally take care of itself.  (Also as an aside it's checking for iis-stop but that id doesn't exist).
```suggestion
```